# GUIA 3

### Miembros
 - Valeria Alvarez
 - Jonathan Alonso
##### Actividad 1. Filtros y cruces

Producto
realice lo siguiente:

- Transformación del tipo de dato.
- Cruce de tablas.
- Filtro de variables por algún tipo de criterio.
- Enviar el archivo en formato HTML al tutor en la plataforma.

Recomendaciones:

- Orientar el acercamiento a los materiales de estudio.
- Asegura coherencia y suficiente detalle en la secuencia instruccional.
- Detalle las características del entregable.
- Incluir en las instrucciones cómo y cuándo utilizar los recursos de aprendizaje.
  - Se recomienda trabajar en parejas para la revisión de las variables en la base de datos dada.
  - Cargar la base de datos dada y empezar a revisar las variables.
  - Determinar el tipo de cada una de las variables.
  - Aplicar las funciones revisadas, según cada tipo de variable.
  - Realizar la limpieza de la variable NIT proveedor.
  - Transformar la variable del Nit.
  - Cruzar las bases de SECOP I con la del SECOP II.
  - Realizar un informe sobre las variables que se exploraron. 

#### Exploracion de datos

In [202]:
#Se importan las librerias de interes
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
#Se leen las bases de datos dadas
df = pd.read_csv('datos_secop_I_modified.csv',delimiter=',')
print(df.shape)
df.head()

(1000, 19)


identificador_paa  codigo_entidad  \
0             303062        28500134   
1             294652       205001279   
2             259436       286571031   
3             263541       205001191   
4             298985       250226019   

                                      nombre_entidad  nit_entidad  anno_paa  \
0                 CASANARE  ESE HOSPITAL DE YOPAL !     891855029      2022   
1  ANTIOQUIA  INSTITUCIN EDUCATIVA DBORA ARANGO P...    900196642      2022   
2  PUTUMAYO  CENTRO EDUCATIVO RURAL SAUDITA  PUER...  846004233-5      2021   
3  ANTIOQUIA  INSTITUCIN EDUCATIVA SAN CRISTBAL  ...    811017472      2021   
4  META  CENTRO EDUCATIVO RURAL JOSE MARIA GUIOTH...  900134298-9      2022   

                                   direccion_entidad telefono_entidad  \
0  Calle 15 N 795 Manzana L via marginal de la selva          6344699   
1                                  Calle 118  103160          5294058   
2                                     VEREDA SAUDITA       3208339785   
3                              Carrera 131 No 65  07          4279180   
4                     Vda Caney Medio Km2CumaralMeta       3013604535   

                                     mision_y_vision  \
0  MisionEl Hospital Regional de la Orinoquia es ...   
1  MISION La Institucin Educativa Dbora Arango de...   
2  MISION El Centro Educativo Rural Saudita garan...   
3                                               2021   
4                                              Misin   

                             perspectiva_estrategica  \
0  el Hospital Regional de la Orinoqua ESE se con...   
1  Direccionar la Institucin para el logro del ho...   
2                                                  0   
3                             wwwiesancristobaleduco   
4  Mejoramiento de la calidad educativa en la pre...   

                                       info_contacto  \
0  Almacen Hospital Regional de la Orinoquia Cont...   
1                       Carlos Enrique Rojas Sanchez   
2                                         3208339785   
3  La  Institucin  Educativa San Cristbal ofrece ...   
4  LUIS FERNANDO ROJAS SALAZAR CC 17267322 Cargo ...   

                                          ppto_total  \
0                                       126898133557   
1                                          $84026639   
2                                         $107700000   
3  La Institucin educativa San Cristbal Lder en l...   
4                                          $22500000   

                                limite_menor_cuantia limite_minima_cuantia  \
0                                                NaN                  $nan   
1                                                  0              20000000   
2                                          254387280             $25438728   
3  JOSE CHARLES VELEZ BETANCURRectoriesancristoba...             159239220   
4                                          280000000             $28000000   

  fecha_ultima_actualizacion             fecha_cargue  municipio_paa  \
0                   12/31/21  2022-01-31T11:53:47.000          Yopal   
1                    12/1/21  2022-01-26T09:43:07.000       Medellín   
2                   11/27/20  2021-01-28T02:34:58.000  Puerto Guzmán   
3                  181705200  2021-01-29T04:15:31.000       Medellín   
4                    12/2/21  2022-01-28T07:06:52.000        Cumaral   

  departamento_paa localidad_paa  codigo_municipio  
0         Casanare    PLAN UNICO             85001  
1        Antioquia    UNICO PLAN              5001  
2         Putumayo    UNICO PLAN             86571  
3        Antioquia    UNICO PLAN              5001  
4             Meta    UNICO PLAN             50226

In [203]:
#Vemos las columnas
df.columns

Index(['identificador_paa', 'codigo_entidad', 'nombre_entidad', 'nit_entidad',
       'anno_paa', 'direccion_entidad', 'telefono_entidad', 'mision_y_vision',
       'perspectiva_estrategica', 'info_contacto', 'ppto_total',
       'limite_menor_cuantia', 'limite_minima_cuantia',
       'fecha_ultima_actualizacion', 'fecha_cargue', 'municipio_paa',
       'departamento_paa', 'localidad_paa', 'codigo_municipio'],
      dtype='object')

In [204]:
#Observamos tipo de datos
df.dtypes

identificador_paa              int64
codigo_entidad                 int64
nombre_entidad                object
nit_entidad                   object
anno_paa                       int64
direccion_entidad             object
telefono_entidad              object
mision_y_vision               object
perspectiva_estrategica       object
info_contacto                 object
ppto_total                    object
limite_menor_cuantia          object
limite_minima_cuantia         object
fecha_ultima_actualizacion    object
fecha_cargue                  object
municipio_paa                 object
departamento_paa              object
localidad_paa                 object
codigo_municipio               int64
dtype: object

In [205]:
#Resumen variables numericas
df.describe()

identificador_paa  codigo_entidad     anno_paa  codigo_municipio
count        1000.000000    1.000000e+03  1000.000000       1000.000000
mean       295795.588000    1.781937e+08  2021.682000      40750.462000
std         25046.125875    1.007772e+08     0.739881      27234.821954
min        219842.000000    2.538170e+05  2020.000000       1101.000000
25%        274957.750000    2.888282e+07  2021.000000      15777.000000
50%        297910.500000    2.176590e+08  2022.000000      41396.000000
75%        318159.000000    2.524203e+08  2022.000000      68307.000000
max        338576.000000    2.990010e+08  2023.000000      99773.000000

In [206]:
#Resumen Variables objeto
df.describe(include='object')

nombre_entidad  nit_entidad  \
count                                       1000         1000   
unique                                       970          946   
top     CAQUET  ALCALDA MUNICIPIO DE EL DONCELLO  No Definido   
freq                                           3           22   

       direccion_entidad telefono_entidad mision_y_vision  \
count               1000             1000             997   
unique               950              920             940   
top          No Definido      No Definido     No Definido   
freq                  19               28              22   

       perspectiva_estrategica info_contacto ppto_total limite_menor_cuantia  \
count                      950           998        998                  990   
unique                     804           919        935                  302   
top                No Definido   No Definido         $0           $280000000   
freq                        45            28         23                   96   

       limite_minima_cuantia fecha_ultima_actualizacion  \
count                    990                        989   
unique                   260                        508   
top                 25438728                    1/31/22   
freq                     102                         41   

                   fecha_cargue municipio_paa departamento_paa localidad_paa  
count                      1000          1000             1000          1000  
unique                      974           511               32           272  
top     2022-01-31T11:53:47.000   Bogotá D.C.     Cundinamarca    UNICO PLAN  
freq                          2            44               93           596

#### Correjimos los tipos de datos
los tipos de datos a cambiar son:
- identificador_paa: str
- codigo_entidad: str
- codigo_municipio: str
- ppto_total: float
- limite_menor_cuantia: float
- limite_minima_cuantia: float
- fecha_ultima_actualizacion: datetime
- fecha_cargue: timestamp

In [207]:
df['identificador_paa']=df['identificador_paa'].astype(str)
df['codigo_entidad']=df['codigo_entidad'].astype(str)
df['identificador_paa']=df['identificador_paa'].astype(str)
df['codigo_municipio']=df['codigo_municipio'].astype(str)
df['ppto_total']= df['ppto_total'].apply(lambda x: x.replace('$','') if isinstance(x, str) else x )
df['limite_menor_cuantia']= df['limite_menor_cuantia'].apply(lambda x: x.replace('$','')  if isinstance(x, str) else x)
df['limite_minima_cuantia']= df['limite_minima_cuantia'].apply(lambda x: x.replace('$','') if isinstance(x, str) else x)

In [208]:
def safe_convert_to_float(value):
    try:
        return float(value)
    except (ValueError, TypeError):
        return None  # O un valor predeterminado de tu elección

In [209]:
df['ppto_total'] = df['ppto_total'].apply(safe_convert_to_float)
df['limite_menor_cuantia']= df['limite_menor_cuantia'].apply(safe_convert_to_float)
df['limite_minima_cuantia']= df['limite_minima_cuantia'].apply(safe_convert_to_float)

In [210]:
def safe_convert_to_date(value):
    try:
        return pd.to_datetime(value,format='%m/%d/%y')
    except:
        try:
            return pd.to_datetime(value,format='%d%m%y')
        except:
            try:
                return pd.to_datetime(value,format='%d de %B de %Y')
            except (ValueError, TypeError):
                return None  # O un valor predeterminado de tu elección

In [211]:
df['fecha_ultima_actualizacion']=df['fecha_ultima_actualizacion'].apply(safe_convert_to_date)

In [212]:
def safe_convert_to_timestamp(value):
    try:
        return pd.Timestamp(value)
    except (ValueError, TypeError):
        return None  # O un valor predeterminado de tu elección

In [213]:
df['fecha_cargue']=df['fecha_cargue'].apply(safe_convert_to_timestamp)

In [214]:
df['nit_entidad']=df['nit_entidad'].apply(lambda x: x.replace('.',''))
df['nit_entidad']=df['nit_entidad'].apply(lambda x: x[:9])
df['nit_entidad']=df['nit_entidad'].apply(safe_convert_to_float)
df['nit_entidad']=df['nit_entidad'].astype(str)

In [215]:
df.describe()

anno_paa    ppto_total  limite_menor_cuantia  limite_minima_cuantia  \
count  1000.000000  9.380000e+02          8.410000e+02           8.460000e+02   
mean   2021.682000  4.374557e+11          7.131683e+09           4.217817e+08   
min    2020.000000  0.000000e+00          0.000000e+00           0.000000e+00   
25%    2021.000000  5.097459e+07          2.457808e+08           2.543873e+07   
50%    2022.000000  1.564156e+08          2.543873e+08           2.800000e+07   
75%    2022.000000  3.073558e+09          2.800000e+08           3.248000e+07   
max    2023.000000  6.279704e+13          2.090174e+12           8.694032e+10   
std       0.739881  3.550660e+12          9.989706e+10           3.862472e+09   

          fecha_ultima_actualizacion                   fecha_cargue  
count                            651                           1000  
mean   2021-09-03 08:02:12.718894080  2021-12-25 13:29:01.853999872  
min              2000-01-20 00:00:00            2020-01-28 09:00:15  
25%              2021-01-29 00:00:00            2021-02-10 07:18:16  
50%              2021-12-03 00:00:00            2022-01-28 03:04:21  
75%              2022-01-31 00:00:00     2022-06-23 01:45:54.500000  
max              2023-12-31 00:00:00            2023-10-31 03:32:25  
std                              NaN                            NaN

#### Informe SECOP I
Se realizó una limpieza preliminar de los datos en la que se identificaron y corrigieron valores erróneos en varias columnas, utilizando funciones de manejo de errores (try-except) para reemplazar los valores incorrectos por NaN y evitar la corrupción de los datos. 

El análisis de la tabla muestra que los datos cubren el período de 2020 a 2023, con una amplia variabilidad en los valores de presupuesto total y límites. Las medias son significativamente mayores que las medianas, lo que indica la presencia de valores extremos altos. La alta dispersión y las elevadas desviaciones estándar sugieren que algunos registros son notablemente altos en comparación con la mayoría. Las fechas de actualización y carga van desde el año 2000 hasta 2023, reflejando un proceso continuo de registro y actualización. Se recomienda realizar un análisis más detallado de los valores atípicos y la distribución para obtener una comprensión más precisa de los datos.


#### Exploramos datos de df2

In [216]:
df2 = pd.read_csv('datos_secov_II_modified.csv',delimiter=';',encoding='latin-1')
df.shape

(1000, 19)

In [217]:
df2.head()

identificador_unico id_plan_anual_de_adquisiciones  anno nit_entidad  \
0     CO1.APP.2041264           MarketplaceCO1538159  2024   721520377   
1     CO1.APP.2041602           MarketplaceCO1538240  2024   718938582   
2     CO1.APP.2041198           MarketplaceCO1538184  2024   703036004   
3     CO1.APP.2041074           MarketplaceCO1538121  2024   723119863   
4     CO1.APP.2041030           MarketplaceCO1415172  2024   705153815   

                                    nombre_entidad  \
0  INSTITUCION EDUCATIVA AGROPECUARIA MAXIMO GOMEZ   
1       INSTITUCION EDUCATIVA MUNICIPAL EL SOCORRO   
2                              MUNICIPIO LA DORADA   
3                  COOAGUAS DE PUEBLO NUEVO E.S.P.   
4                            MUNICIPIO DE LA PLATA   

                                       mision_vision  \
0  MISION 2023 2026 La Institucin Educativa Agrop...   
1  MISION Brindar una formacin acadmica que forme...   
2  La administracin municipal regir los destinos ...   
3  MISION LA ADMINISTRACION PBLICA COOPERATIVA EM...   
4  Gestionar servir a la comunidad promover el bi...   

                             perspectiva_estrategica  \
0  Ser una institucin destacada por sus estrategi...   
1  El proceso de formacin de los educandos debe e...   
2  La Dorada ser un municipio competitivo y soste...   
3  POLITICA DE CALIDAD Somos un organismo pblico ...   
4  A travs de la implementacin del presente Plan ...   

                 nombre_contacto telefono_contacto  \
0  MARIA EUGENIA LEDESMA LLANTEN        3104661182   
1        RAFAEL BASTIDAS SANCHEZ        3136820189   
2  JULIAN DAVID OLIVEROS SERRANO           8572013   
3   CARLOS FERNANDO GERMAN CALAO        3127695263   
4      CRISTINA HOLGUIN GRISALES        3223001245   

                  correo_contacto  ... presupuesto_menor_cuantia  \
0  mariaeugenia.maestra@gmail.com  ...               $26000000.0   
1        rafabastidas@hotmail.com  ...              $845000000.0   
2  compras@ladorada-caldas.gov.co  ...              $364000000.0   
3        carlosgcalao@outlook.com  ...               174000000.0   
4    almacen@laplata-huila.gov.co  ...              $364000000.0   

  presupuesto_minima_cuantia         fecha_de_publicacion  \
0                 $2600000.0  2024-07-31 03:47:42.6041815   
1                $84500000.0  2024-01-31 16:19:08.5384812   
2                $36400000.0  2024-01-17 20:54:03.3001599   
3                116000000.0  2024-02-01 02:01:48.3772533   
4                $36400000.0  2024-01-10 15:55:50.6223556   

             fecha_de_creacion   fecha_de_ultima_modificacion version  \
0  2024-08-23 07:11:00.0000000  2024-08-22 20:32:26.391947400       2   
1  2024-08-23 07:11:00.0000000  2024-08-22 22:00:52.607658300      13   
2  2024-08-23 07:11:00.0000000  2024-08-22 21:03:45.672865800      28   
3  2024-08-23 07:11:00.0000000  2024-08-22 19:47:30.908245300       2   
4  2024-08-23 07:11:00.0000000  2024-08-22 15:48:03.701324800     148   

   identificador_requerimiento_compra           estado departamento_paa  \
0                    CO1.PPI.33807381         Borrador            Cauca   
1                    CO1.PPI.33809817  No especificado           Nariño   
2                    CO1.PPI.33808340         Borrador           Caldas   
3                    CO1.PPI.33805980         Borrador          Córdoba   
4                    CO1.PPI.33799556        Publicado            Huila   

  municipio_paa  
0       Cajibío  
1         Pasto  
2     La Dorada  
3  Pueblo Nuevo  
4      La Plata  

[5 rows x 21 columns]

In [218]:
df2.describe()

anno      version
count  1000.000000  1000.000000
mean   2022.984000    33.862000
std       1.685997    66.973774
min    2016.000000     1.000000
25%    2023.000000     3.000000
50%    2024.000000     9.000000
75%    2024.000000    37.250000
max    2025.000000   818.000000

In [219]:
df.describe(include='object')

identificador_paa codigo_entidad  \
count               1000           1000   
unique               974            941   
top               303062      218247011   
freq                   2              3   

                                  nombre_entidad nit_entidad  \
count                                       1000        1000   
unique                                       970         882   
top     CAQUET  ALCALDA MUNICIPIO DE EL DONCELLO         nan   
freq                                           3          50   

       direccion_entidad telefono_entidad mision_y_vision  \
count               1000             1000             997   
unique               950              920             940   
top          No Definido      No Definido     No Definido   
freq                  19               28              22   

       perspectiva_estrategica info_contacto municipio_paa departamento_paa  \
count                      950           998          1000             1000   
unique                     804           919           511               32   
top                No Definido   No Definido   Bogotá D.C.     Cundinamarca   
freq                        45            28            44               93   

       localidad_paa codigo_municipio  
count           1000             1000  
unique           272              524  
top       UNICO PLAN             1101  
freq             596               44

In [220]:
df2.dtypes

identificador_unico                   object
id_plan_anual_de_adquisiciones        object
anno                                   int64
nit_entidad                           object
nombre_entidad                        object
mision_vision                         object
perspectiva_estrategica               object
nombre_contacto                       object
telefono_contacto                     object
correo_contacto                       object
valor_presupuesto_general             object
presupuesto_menor_cuantia             object
presupuesto_minima_cuantia            object
fecha_de_publicacion                  object
fecha_de_creacion                     object
fecha_de_ultima_modificacion          object
version                                int64
identificador_requerimiento_compra    object
estado                                object
departamento_paa                      object
municipio_paa                         object
dtype: object

#### Correjimos los tipos de datos
los tipos de datos a cambiar son:

- valor_presupuesto_general: float 
- presupuesto_menor_cuantia: float
- presupuesto_minima_cuantia: float
- fecha_de_publicacion: timestamp                
- 
fecha_de_creacio: timestamp                 t- 
fecha_de_ultima_modificaci: timestampn       ct

In [221]:
df2['valor_presupuesto_general']=df2['valor_presupuesto_general'].apply(lambda x: x.replace('$','') if isinstance(x, str) else x )
df2['presupuesto_menor_cuantia']=df2['presupuesto_menor_cuantia'].apply(lambda x: x.replace('$','') if isinstance(x, str) else x )
df2['presupuesto_minima_cuantia']=df2['presupuesto_minima_cuantia'].apply(lambda x: x.replace('$','') if isinstance(x, str) else x )

In [222]:
df2['fecha_de_publicacion']=df2['fecha_de_publicacion'].apply(safe_convert_to_timestamp)
df2['fecha_de_creacion']=df2['fecha_de_creacion'].apply(safe_convert_to_timestamp)
df2['fecha_de_ultima_modificacion']=df2['fecha_de_ultima_modificacion'].apply(safe_convert_to_timestamp)
df2['nit_entidad']=df2['nit_entidad'].apply(lambda x: x[:9])

#### Informe SECOP II
Se limpiaron los datos monetarios para permitir un análisis más preciso y se convirtieron las fechas a formato timestamp.

La tabla muestra una amplia variabilidad en los datos de las entidades, con numerosos valores no definidos en campos clave como dirección, teléfono, y misión y visión. Los años de los registros se extienden desde 2016 hasta 2025, con un promedio cercano a 2023. Las versiones de los datos abarcan un rango amplio, desde valores mínimos de 1 hasta máximos de 818, con una alta dispersión. Se recomienda una revisión más detallada de los valores no definidos y de la variabilidad en las versiones para mejorar la calidad y precisión de la información.

In [223]:
df.merge(df2, on='nit_entidad')

Empty DataFrame
Columns: [identificador_paa, codigo_entidad, nombre_entidad_x, nit_entidad, anno_paa, direccion_entidad, telefono_entidad, mision_y_vision, perspectiva_estrategica_x, info_contacto, ppto_total, limite_menor_cuantia, limite_minima_cuantia, fecha_ultima_actualizacion, fecha_cargue, municipio_paa_x, departamento_paa_x, localidad_paa, codigo_municipio, identificador_unico, id_plan_anual_de_adquisiciones, anno, nombre_entidad_y, mision_vision, perspectiva_estrategica_y, nombre_contacto, telefono_contacto, correo_contacto, valor_presupuesto_general, presupuesto_menor_cuantia, presupuesto_minima_cuantia, fecha_de_publicacion, fecha_de_creacion, fecha_de_ultima_modificacion, version, identificador_requerimiento_compra, estado, departamento_paa_y, municipio_paa_y]
Index: []

[0 rows x 39 columns]

#### Conclusión Actividad I
la base de datos SECOP I y SECOP II no comparten registros en común, por lo cual procedermos a realizar concatenación de datos y realizar la revisión de la información pertinente

In [240]:
#Seleccionamos las columnas de interes
secop2=df2[['anno', 'departamento_paa','fecha_de_creacion','fecha_de_publicacion','fecha_de_ultima_modificacion',
 'municipio_paa','nit_entidad','nombre_entidad','presupuesto_menor_cuantia','presupuesto_minima_cuantia',
 'telefono_contacto','valor_presupuesto_general']]

In [243]:
#filtramos df con las columnas equivalentes escogidas para secop2
secop1=df[['anno_paa','departamento_paa','fecha_cargue','fecha_ultima_actualizacion', 'municipio_paa',
            'nit_entidad','nombre_entidad','limite_menor_cuantia','limite_minima_cuantia','telefono_entidad','ppto_total']]
#renombramos las columnas para poder concatenar
secop1=secop1.rename(columns={"anno_paa": "anno", "fecha_cargue": "fecha_de_creacion",
                      "fecha_ultima_actualizacion":"fecha_de_ultima_modificacion",
                      "limite_menor_cuantia":"presupuesto_menor_cuantia","limite_minima_cuantia":"presupuesto_minima_cuantia",
                      "telefono_entidad":"telefono_contacto","ppto_total":"valor_presupuesto_general"})
#Agregamos la columna faltante
secop1['fecha_de_publicacion']=secop1['fecha_de_creacion']

In [248]:
#concatenamos las tablas filtradas
secop=pd.concat([secop2,secop1], ignore_index=True)

In [250]:
#Creamos una tabla con 2000 registros
secop

anno departamento_paa   fecha_de_creacion          fecha_de_publicacion  \
0     2024            Cauca 2024-08-23 07:11:00 2024-07-31 03:47:42.604181500   
1     2024           Nariño 2024-08-23 07:11:00 2024-01-31 16:19:08.538481200   
2     2024           Caldas 2024-08-23 07:11:00 2024-01-17 20:54:03.300159900   
3     2024          Córdoba 2024-08-23 07:11:00 2024-02-01 02:01:48.377253300   
4     2024            Huila 2024-08-23 07:11:00 2024-01-10 15:55:50.622355600   
...    ...              ...                 ...                           ...   
1995  2021        Antioquia 2021-01-31 08:01:37 2021-01-31 08:01:37.000000000   
1996  2021          Caquetá 2021-04-15 10:49:31 2021-04-15 10:49:31.000000000   
1997  2022        Antioquia 2022-11-29 11:31:24 2022-11-29 11:31:24.000000000   
1998  2022            Huila 2022-07-14 00:01:24 2022-07-14 00:01:24.000000000   
1999  2022            Cesar 2022-01-28 04:13:09 2022-01-28 04:13:09.000000000   

      fecha_de_ultima_modificacion municipio_paa  nit_entidad  \
0    2024-08-22 20:32:26.391947400       Cajibío    721520377   
1    2024-08-22 22:00:52.607658300         Pasto    718938582   
2    2024-08-22 21:03:45.672865800     La Dorada    703036004   
3    2024-08-22 19:47:30.908245300  Pueblo Nuevo    723119863   
4    2024-08-22 15:48:03.701324800      La Plata    705153815   
...                            ...           ...          ...   
1995 2020-11-30 00:00:00.000000000      Medellín  811017836.0   
1996                           NaT        Solano  900155525.0   
1997 2022-10-01 00:00:00.000000000       Uramita  890984575.0   
1998                           NaT         Neiva  891180005.0   
1999                           NaT     Aguachica  812001531.0   

                                         nombre_entidad  \
0       INSTITUCION EDUCATIVA AGROPECUARIA MAXIMO GOMEZ   
1            INSTITUCION EDUCATIVA MUNICIPAL EL SOCORRO   
2                                   MUNICIPIO LA DORADA   
3                       COOAGUAS DE PUEBLO NUEVO E.S.P.   
4                                 MUNICIPIO DE LA PLATA   
...                                                 ...   
1995  ANTIOQUIA  INSTITUCIN EDUCATIVA SANTA ELENA  M...   
1996                   CAQUET  CE PEÑAS BLANCAS  SOLANO   
1997  ANTIOQUIA  PERSONERIA MUNICIPAL DE URAMITA  UR...   
1998   HUILA  INSTITUCIN EDUCATIVA SANTA LIBRADA  NEIVA   
1999  CESAR  INSTITUTO MUNICIPAL DE DEPORTES RECREAC...   

     presupuesto_menor_cuantia presupuesto_minima_cuantia telefono_contacto  \
0                   26000000.0                  2600000.0        3104661182   
1                  845000000.0                 84500000.0        3136820189   
2                  364000000.0                 36400000.0           8572013   
3                  174000000.0                116000000.0        3127695263   
4                  364000000.0                 36400000.0        3223001245   
...                        ...                        ...               ...   
1995                17556040.0                 24578456.0           5381304   
1996                       NaN                        NaN        3209439424   
1997               254387280.0                 25250000.0      317 400 4524   
1998               245784840.0                 24578484.0           8750407   
1999               280000000.0                 28000000.0           5662167   

     valor_presupuesto_general  
0                   98097506.0  
1                  394955722.0  
2                81756864004.0  
3                 1006390000.0  
4                17288164198.0  
...                        ...  
1995                92111926.0  
1996                       NaN  
1997                14500000.0  
1998               108416516.0  
1999              1342639796.0  

[2000 rows x 12 columns]

#### Actividad 2. Visualizaciónes seleccionadas.
 

Realice lo siguiente:

 - Identificación y alguna transformación del tipo de dato.
 - Filtro de variables por algún tipo de criterio.
 - Utilizar al menos 2 tipos de gráficos para variables seleccionadas. 
 - Enviar el archivo en formato HTML al tutor en la plataforma.

Instrucciones específicas para desarrollar el producto

 - Se recomienda trabajar en parejas para la revisión de las variables en la base de datos dada.
 - Cargar la base de datos dada y empezar a revisar las variables.
 - Determinar el tipo de cada una de las variables.
 - Aplicar las funciones revisadas, para la transformación del dato y cambiar su tipo.
 - Realizar filtros de tablas requeridos.
 - Realizar las gráficas de algunas variables seleccionadas.
 